drv/lpc55-swd/src/armv7debug.rs

```diff
@@ -0,0 +1,189 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+//
+//
+// For the STM32H7xx, see ARMv7-M Architecture Reference Manual
+// Part 3 Debug Arch.
+// https://developer.arm.com/documentation/ddi0403/d/Debug-Architecture?lang=en
+
+use bitflags::bitflags;
+
+pub trait DpAddressable {
+    // Address for accessing a DP Register
+    fn addr() -> u32;
+}
```

Comment:
Nit: you can put the `ADDRESS` constant into the trait, e.g.
```suggestion
pub trait DpAddressable {
    /// Address for accessing a DP Register
    const ADDRESS: u32;
}
```